### DISCUSSION TITLE
0.1.0-beta Release

### DISCUSSION THREAD

**Discussion by jessdtate** [SCIInstitute/UncertainSCI#80] [General]
The beta release for UncertainSCI is here.  This version includes some bug fixes from the alpha release, a new feature to adaptively find the optimal order of the problem, and some new examples to try.  

<hr /><em>This discussion was created from the release <a href='https://github.com/SCIInstitute/UncertainSCI/releases/tag/0.1.0-beta'>0.1.0-beta Release</a>.</em>

**Comment by ww382** [comment]
Hi! 

I downloaded the beta version of the source code and installed UncertainSCI following the instructions [here](https://uncertainsci.readthedocs.io/en/latest/user_docs/getting_started.html#installing-uncertainsci). 

When I proceeded to run demo/build_pce.py. I received an error message indicating 

 Traceback (most recent call last):
  File "../demos/build_pce.py", line 10, in <module>
    from UncertainSCI.distributions import BetaDistribution
ModuleNotFoundError: No module named 'UncertainSCI'

I would appreciate if you can provide me suggestions on how to resolve this error. 

For your reference, I listed the packages that were installed in my conda environment below. 

**Package         Version**
certifi                    2020.4.5.1
cycler                    0.10.0
kiwisolver              1.2.0
matplotlib             3.1.3
numpy                   1.17.2
packaging             20.3
pip                         21.0.1
pyparsing             2.4.7
python-dateutil    2.8.1
scipy                     1.4.1
setuptools            58.0.4
six                         1.15.0
UncertainSCI       0.0.0
wheel                   0.37.0

**Reply by jessdtate** [reply]
Thanks for your question.  I wonder if this is a path problem.  Try adding the `UncertainSCI/UncertainSCI` to your python path
```
import sys
sys.path.append('path')
```
You can this to the script, or modify the PYTHONPATH environment variable, which is system dependent.

**Reply by ww382** [reply]
Thank you so much! Adding the path resolved the ModuleNotFoundError. 

Now I received a segmentation fault error: 

Directory data_set created
Precomputing data for Jacobi parameters (alpha,beta) = (0.000000, -0.500000)...Done
This will query the model 66 times
zsh: segmentation fault  python build_pce.py

**Reply by jessdtate** [reply]
Are you running on big sur, or what operating system?

**Reply by ww382** [reply]
yes, I'm running on big sur version 11.5.2.

**Reply by jessdtate** [reply]
There is a know bug with matplotib and big sur.  You can comment out the plotting function and the script will finish.  I'll check the status on that bug and see when it might be fixed

**Reply by jessdtate** [reply]
Looks like there might be a fix by uninstalling and reinstalling matplotlib. 

https://code2care.org/python/python-matplotib-segmentation-flult-11-macos-big-sur

**Reply by ww382** [reply]
uninstalling and reinstalling matplotlib works! Thank you so much!

**Comment by ww382** [comment]
I've been running the demo scripts and trying to understand how UncertainSCI works. How do I output the Sobol sensitivity indices, Np, and Ns as shown in, for example, Table A1 in Burk et al? 

Burk, KM, Narayan, A, Orr, JA. Efficient sampling for polynomial chaos-based uncertainty quantification and sensitivity analysis using weighted approximate Fekete points. Int J Numer Meth Biomed Engng. 2020; 36:cnm03395.
